src/widgets/search-box/search-box.js

```diff
@@ -124,7 +124,24 @@ function searchBox({
       // Add all the needed attributes and listeners to the input
       this.addDefaultAttributesToInput(input, state.query);
 
-      // Keep keyup to handle searchOnEnterKeyPressOnly
+      // like jquery closest()
+      function closest(el, selector) {
+        let matches = el.webkitMatchesSelector ? 'webkitMatchesSelector' : (el.msMatchesSelector ? 'msMatchesSelector' : 'matches');
+        while (el.parentElement) {
```

Comment:
why not just smg like while(parent !== FORM) parent=parentNode?

dunno msMatchesSelector will have to look